gothemis/session/session.go

```diff
@@ -37,10 +37,23 @@ const (
 
 // Errors returned by Secure Session.
 var (
-	ErrMissingClientID   = errors.NewWithCode(errors.InvalidParameter, "empty client ID for Secure Session")
-	ErrMissingPrivateKey = errors.NewWithCode(errors.InvalidParameter, "empty client private key for Secure Session")
-	ErrMissingMessage    = errors.NewWithCode(errors.InvalidParameter, "empty message for Secure Session")
-	ErrOverflow          = errors.NewWithCode(errors.NoMemory, "Secure Session cannot allocate enough memory")
+	ErrDestroyObj                = errors.New("Failed to destroy secure session object")
+	ErrCreateObj                 = errors.New("Failed to create secure session object")
+	ErrGetRequestSize            = errors.New("Failed to get request size")
+	ErrGenRequest                = errors.New("Failed to generate request")
+	ErrGetWrappedSize            = errors.New("Failed to get wrapped size")
+	ErrWrapData                  = errors.New("Failed to wrap data")
+	ErrGetUnwrappedSize          = errors.New("Failed to get unwrapped size")
+	ErrUnwrapData                = errors.New("Failed to unwrap data")
+	ErrCallbackGetUnwrappedSize  = errors.NewCallbackError("Failed to get unwraped size (get_public_key_by_id callback error)")
+	ErrCallbackUnwrapData        = errors.NewCallbackError("Failed to unwrap data (get_public_key_by_id callback error)")
+	ErrCallbackGetRemIDLen       = errors.NewCallbackError("Failed to get session remote id length")
+	ErrCallbackBadRemIDLen       = errors.NewCallbackError("Incorrect remote id length (0)")
+	ErrCallbackGetRemID          = errors.NewCallbackError("Failed to get session remote id")
+	ErrMissingClientID           = errors.NewWithCode(errors.InvalidParameter, "empty client ID for Secure Session")
+	ErrMissingPrivateKey         = errors.NewWithCode(errors.InvalidParameter, "empty client private key for Secure Session")
+	ErrMissingMessage            = errors.NewWithCode(errors.InvalidParameter, "empty message for Secure Session")
+	ErrOverflow                  = errors.NewWithCode(errors.NoMemory, "Secure Session cannot allocate enough memory")
```

Comment:
Not clear what exactly overflow. Memory, int capacity, buffer capacity, amount of requests.... Let's give the name which enough just to read and it will describe what exactly overflowed?)

Reply:
What about `ErrOverflow` -> `ErrOutOfMemory`? Quite common name for such types of errors

Reply:
Yeah, it is much better. Advice for the future. It's good to search similar cases in the other code of project or in similar projects to not introduce something new and use the most common names/patterns/etc. Probably we used some errors like that in other parts of project)

Reply:
Yeah, all these checks are for size overflow, but from Go viewpoint this can be treated as out-of-memory error. Themis Core asks Go to allocate a ginormous buffer with a length that can't even be represented with Go types, so Go cannot allocate that many.